From: Andrew Chadwick <a.t.chadwick@gmail.com>
Date: Wed, 14 Dec 2016 18:20:23 +0000
Subject: Remove references to BOGOSITY cursor

It may not exist on newer systems/themes, and may cause exceptions to be
thrown in Wayland. We're only using it as an "obvious" fallback anyway.

Addresses mypaint/mypaint#750.

[Cherry-pick of 2c07a27b00c421979cda17d7827fce85d276c28b from master]

Origin: upstream, 1.2.1
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=859064
---
 gui/cursor.py | 4 ++--
 gui/mode.py   | 6 ++++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/gui/cursor.py b/gui/cursor.py
index df8b33a..5c24d4f 100644
--- a/gui/cursor.py
+++ b/gui/cursor.py
@@ -305,10 +305,10 @@ class CustomCursorMaker (object):
         # Find a small action icon for the overlay
         action = self.app.find_action(action_name)
         if action is None:
-            return Gdk.Cursor.new(Gdk.CursorType.BOGOSITY)
+            return Gdk.Cursor.new(Gdk.CursorType.ARROW)
         icon_name = action.get_icon_name()
         if icon_name is None:
-            return Gdk.Cursor.new(Gdk.CursorType.BOGOSITY)
+            return Gdk.Cursor.new(Gdk.CursorType.ARROW)
         return self.get_icon_cursor(icon_name, cursor_name)
 
     def get_icon_cursor(self, icon_name, cursor_name=Name.ARROW):
diff --git a/gui/mode.py b/gui/mode.py
index 4f9121c..2802720 100644
--- a/gui/mode.py
+++ b/gui/mode.py
@@ -810,7 +810,8 @@ class SingleClickMode (InteractionMode):
     """Base class for non-drag (single click) modes"""
 
     #: The cursor to use when entering the mode
-    cursor = gdk.Cursor(gdk.BOGOSITY)
+    # FIXME: Use Gdk.Cursor.new_for_display; read-only property
+    cursor = Gdk.Cursor.new(Gdk.CursorType.ARROW)
 
     def __init__(self, ignore_modifiers=False, **kwds):
         super(SingleClickMode, self).__init__(**kwds)
@@ -858,7 +859,8 @@ class DragMode (InteractionMode):
 
     """
 
-    inactive_cursor = gdk.Cursor(gdk.BOGOSITY)
+    # FIXME: Use Gdk.Cursor.new_for_display; read-only property
+    inactive_cursor = Gdk.Cursor.new(Gdk.CursorType.ARROW)
     active_cursor = None
 
     #: If true, exit mode when initial modifiers are released
